src/main/java/com/wavefront/sdk/direct/ingestion/WavefrontDirectIngestionClient.java

```diff
@@ -175,8 +193,55 @@ public void flush() throws IOException {
 
   private void internalFlush(LinkedBlockingQueue<String> buffer, String format)
       throws IOException {
+    for (List<String> batch : getBatches(buffer)) {
+      internalFlushBatch(buffer, format, batch);
+    }
+  }
 
-    List<String> batch = getBatch(buffer);
+  private List<List<String>> getBatches(LinkedBlockingQueue<String> buffer) {
+    List<String> block = getBlock(buffer);
+
+    if (messageSizeBytes == null) {
+      return Collections.singletonList(block);
+    }
+
+    List<List<String>> batches = new ArrayList<>();
+    List<String> batch = new ArrayList<>();
+    int numBytesInBatch = 0;
+
+    for (String item : block) {
```

Comment:
if a user has set `messageSizeBytes`, why do we first break the data into a `block` and then break it down further into `messageSizeBytes` chunks? Doing this, we are traversing the data twice, once when we copy it over to a temp buffer that `getBlock` returns and once when we break it down further into `messageSizeBytes` chunks. 

If `messageSizeBytes` is set, can't we just use the `buffer` object to populate `messageSizeBytes` chunks?

Reply:
Consolidated the processing in `getBatch`.